contract/bridge/contracts/VaultReward.sol

```diff
@@ -0,0 +1,206 @@
+  event ClaimRewards(address indexed vaultId, uint256 amount, uint256 claimAt);
+  event ClaimRewardsAndLock(address indexed vaultId, uint256 amount, uint256 claimAt);
+  event UpdateVaultAccClaimableRewards(
+    address indexed vaultId,
+    uint256 oldAccClaimableRewards,
+    uint256 newAccClaimableRewards,
+    uint256 updatedAt
+  );
+
+  modifier onlyOneBtc() {
+    require(msg.sender == oneBtc, "Only OneBtc");
+    _;
+  }
+
+  modifier vaultExist(address _vaultId) {
+    (uint256 btcPublicKeyX,,,,,,,,) = IVaultRegistry(oneBtc).getVault(_vaultId);
+    require(btcPublicKeyX != 0, "Vault does not exist");
+    _;
+  }
+
+  receive() external payable {}
+
+  function initialize(address _oneBtc, address _vaultReserve) public initializer {
+    oneBtc = _oneBtc;
+    vaultReserve = _vaultReserve;
+  }
+
+  function extendVaultLockPeriod(address _vaultId, uint256 _lockPeriod) external vaultExist(_vaultId) {
+    require(_vaultId == msg.sender, "Invalid vaultId");
+
+    // check if the lockPeriod is valid
+    require(_lockPeriod == 3 || _lockPeriod == 6 || _lockPeriod == 12, "Lock period should be one of 3, 6, 12");
+
+    // update vault accClaimableRewards
+    _updateVaultAccClaimableRewards(_vaultId);
+
+    // get vault
+    LockedVault storage vault = lockedVaults[_vaultId];
+
+    // get the current vault lock period
+    uint256 oldLockPeriod = vault.lockPeriod;
+
+    // update the vault lock info
+    uint256 secPerDay = 60 * 60 * 24;
+    uint256 lockPeriodInSec = secPerDay.mul(30).mul(_lockPeriod);
+    if (vault.lockExpireAt < block.timestamp) {   // new or expired vault
+      vault.lockStartAt = block.timestamp;
+      vault.lockPeriod = _lockPeriod;
+      vault.lockExpireAt = block.timestamp.add(lockPeriodInSec);
+      vault.rewardClaimAt = block.timestamp;
+      vault.collateralUpdatedAt = block.timestamp;
+      vault.accClaimableRewards = 0;
+    } else {    // locked vault
+      vault.lockPeriod = vault.lockPeriod.add(_lockPeriod);
+      vault.lockExpireAt = vault.lockExpireAt.add(lockPeriodInSec);
+    }
+
+    emit ExtendVaultLockPeriod(_vaultId, oldLockPeriod, vault.lockPeriod);
+  }
+
+  function _updateVaultAccClaimableRewards(address _vaultId) internal {
+    // get vault
+    LockedVault storage vault = lockedVaults[_vaultId];
+
+    // store the old accClaimableRewards
+    uint256 oldAccClaimableRewards = vault.accClaimableRewards;
+
+    // update the vault info
+    (uint256 claimableRewards, uint256 rewardClaimAt) = getClaimableRewards(_vaultId);
+    vault.accClaimableRewards = claimableRewards;
+    vault.rewardClaimAt = rewardClaimAt;
+
+    emit UpdateVaultAccClaimableRewards(_vaultId, oldAccClaimableRewards, vault.accClaimableRewards, rewardClaimAt);
+  }
+
+  function updateVaultAccClaimableRewards(address _vaultId) external onlyOneBtc {
+    if (block.timestamp <= lockedVaults[_vaultId].lockExpireAt) {
```

Comment:
This function seems safe to allow anyone to call.

Reply:
Agree.